src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -46,52 +34,44 @@ public static function getSubscribedEvents()
     {
         return array(
             Events::PERSIST => 'handlePersist',
-            Events::HYDRATE => 'handleHydrate',
+            Events::METADATA_LOAD => 'handleMetadataLoad',
         );
     }
 
+    public function handleMetadataLoad(MetadataLoadEvent $event)
+    {
+        $metadata = $event->getMetadata();
+
+        if (false === $metadata->getReflectedClass()->isSubclassOf(OrderBehavior::class)) {
+            return;
+        }
+
+        $metadata->addFieldMapping('suluOrder', array(
+            'encoding' => 'system',
+            'property' => self::FIELD,
+        ));
```

Comment:
This method will ultimately only be called when the Metadata is loaded for the first time. We do not currently have a cache, but when we do these methods will not be called at all for most requests, and some subscribers now implement ONLY this event.